### ISSUE TITLE
terminology: straighten out usage of "flow"

### ISSUE THREAD

**Issue by oliver-sanders** [cylc/cylc-flow#4786]
Since we originally renamed this project to "Cylc Flow" our usage of "flow" has changed.

tldr:

* "Workflow" is the thing configured by the `flow.cylc` file.
* "Flows" are logical runs of the workflow graph.

> Some related context:
>
> * https://github.com/cylc/cylc-flow/issues/3817#issuecomment-717206611
> * https://github.com/cylc/cylc-doc/pull/224
> * https://github.com/cylc/cylc-flow/issues/4686

We have mostly straightened out the terminology now but a few things remain:

* [x] The `flow.cylc` file
  * Should really be `workflow.cylc`, we shortened it to `flow.cylc` to be shorter.
  * **Suggest we live with this**, it's not that big an issue.
  * The `flow.cylc` defines the graph the "flows" run so it does make sense.
* [x] `log/flow-config`
  * I think we rejected the name `workflow-config`.
  * Given that the filename is `flow.cylc` perhaps we should **keep this the way it is** too.
* [x] `cylc install --flow-name` (#4823)
  * See https://github.com/cylc/cylc-flow/issues/4741#issuecomment-1082768766
  * This is now quite confusing, especially as we can now add metadata to flows.
  * **Suggest changing to `--id`.**
* [x] `--reflow`
  * **Agreed to retire** term with #4686

Questions:

1) Anything missing from the list?
2) Propose leaving the first two (flow.cylc and flow-config) as is, but changing the latter two. Agreed?

**Pull requests welcome!**

**Comment by dpmatthews** [comment]
> Suggest changing to `--id`

That would be confusing as well, see https://github.com/cylc/cylc-flow/pull/4455.
I'd suggest `--workflow-name` with `-n` as a short form.

**Comment by oliver-sanders** [comment]
> That would be confusing as well, see https://github.com/cylc/cylc-flow/pull/4455.

Yes, the `--id` is only the "full" ID when used in combination with `--no-run-name`, however, we could document that away e.g. `--id, --id-prefix`.

**Comment by dpmatthews** [comment]
> I'd prefer /log/workflow-config for this. We probably rejected that name in the past before the word flow got more specialized? It's not inconsistent with the first point: flow.cylc is the workflow config file; but flow-config is too close to the English (i.e. "flow config" sounds like config for a specific flow, which it isn't). Also, no one will be relying on the current filename yet.

Discussion on this point moved to #4802

**Comment by hjoliver** [comment]
> Discussion on this point moved to https://github.com/cylc/cylc-flow/issues/4802

I like plain `config` as in #4802 :+1:

**Comment by hjoliver** [comment]
I've checked all the check-boxes above - no remaining questions. This can be closed after the various lined PRs are merged.

**Comment by hjoliver** [comment]
TODO: check cylc-doc for any reference to the old terms

**Comment by oliver-sanders** [comment]
Summary:

* flow.cylc - name kept
* flow-config - to be changed, bumped to https://github.com/cylc/cylc-flow/issues/4802
* `cylc install --flow-name` - changed to `--workflow-name`
* `--reflow` - changed to `--flow=new`
* Checked docs for references to `--reflow` and `--flow-name` - found none.
